crates/router/src/cache.rs

```diff
@@ -131,17 +135,25 @@ mod cache_tests {
     }
 
     #[tokio::test]
-    async fn eviction_on_time_test() {
-        let cache = Cache::new(2, 2, None);
+    async fn eviction_on_size_test() {
+        let cache = Cache::new(2, 2, Some(0));
         cache.push("key".to_string(), "val".to_string()).await;
-        tokio::time::sleep(std::time::Duration::from_secs(3)).await;
         assert_eq!(cache.get_val::<String>("key"), None);
     }
 
     #[tokio::test]
-    async fn eviction_on_size_test() {
-        let cache = Cache::new(2, 2, Some(0));
+    async fn invalidate_cache_for_key() {
+        let cache = Cache::new(1800, 1800, None);
         cache.push("key".to_string(), "val".to_string()).await;
+
+        cache.remove("key").await;
```

Comment:
Can you add the assert it with for value present since this is a test.

Reply:
the changes are pushed now.